base/array.jl

```diff
@@ -664,6 +664,14 @@ end
 _array_for(::Type{T}, itr, ::HasLength) where {T} = Array{T,1}(Int(length(itr)::Integer))
 _array_for(::Type{T}, itr, ::HasShape) where {T} = similar(Array{T}, indices(itr))
 
+function _isconcreteunion(::Type{T}) where T
+    if T isa Union
+        _isconcreteunion(T.a) && _isconcreteunion(T.b)
+    else
+        isconcrete(T)
+    end
+end
+
```

Comment:
We already have `Base.isbitsunion` for this.

Reply:
This one is slightly different, it also includes things like `Union{String, Void}`.